cmd/stopScheduledJobs.go

```diff
@@ -0,0 +1,180 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+
+	"github.com/equinor/radix-cli/generated/radixapi/client/job"
+	"github.com/equinor/radix-cli/pkg/client"
+	"github.com/equinor/radix-cli/pkg/config"
+	"github.com/equinor/radix-cli/pkg/flagnames"
+	"github.com/equinor/radix-cli/pkg/utils/completion"
+	"github.com/spf13/cobra"
+)
+
+// stopComponentCmd represents the stop component command
+var stopScheduledJobsCmd = &cobra.Command{
+	Use:   "scheduled-job",
+	Short: "Stop jobs",
+	Long: `Stop one or all scheduled jobs
+  - Stops scheduled jobs for a job component`,
+	Example: `
+	  # Stop all scheduled single jobs for a job component
+	  rx stop scheduled-job --application radix-test --environment dev --component my-job-component --all-jobs
+
+	  # Stop a scheduled job for a job component
+	  rx stop scheduled-job --application radix-test --environment dev --component my-job-component --job an-unique-single-job-name
+
+	  # Stop all scheduled batches for a job component
+	  rx stop scheduled-job --application radix-test --environment dev --component my-job-component --all-batches
+
+	  # Stop a scheduled batch for a job component
+	  rx stop scheduled-job --application radix-test --environment dev --component my-job-component --batch an-unique-batch-name
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		appName, err := config.GetAppNameFromConfigOrFromParameter(cmd, flagnames.Application)
+		if err != nil {
+			return err
+		}
+
+		envName, err := cmd.Flags().GetString(flagnames.Environment)
+
+		if err != nil || appName == "" || envName == "" {
+			return errors.New("environment name and application name are required fields")
+		}
+
+		cmpName, err := cmd.Flags().GetString(flagnames.Component)
+		if err != nil {
+			return errors.New("job component name is a required field")
+		}
+
+		batchName, err := cmd.Flags().GetString(flagnames.Batch)
+		if err != nil {
+			return err
+		}
+		jobName, err := cmd.Flags().GetString(flagnames.Job)
+		if err != nil {
+			return err
+		}
+		if len(batchName) > 0 && len(jobName) > 0 {
+			return errors.New("batch and job names cannot be used at the same time")
+		}
+		allJobs, err := cmd.Flags().GetBool(flagnames.AllJobs)
+		if err != nil {
+			return err
+		}
+		allBatches, err := cmd.Flags().GetBool(flagnames.AllBatches)
+		if err != nil {
+			return err
+		}
+		if allJobs && allBatches {
+			return errors.New("options --all-jobs and --all-batches cannot be used at the same time")
+		}
+
+		cmd.SilenceUsage = true
+
+		if jobName == "" && batchName == "" {
+			if allJobs {
+				return stopAllJobs(cmd, appName, envName, cmpName)
+			} else if allBatches {
+				return stopAllBatches(cmd, appName, envName, cmpName)
+			}
+			return errors.New("when batch and/or job name are not specified, options --all-jobs or --all-batches are expected")
```

Comment:
Could we allow specifying --all-jobs and --allbatches at the same time?

```suggestion
			if allJobs {
				if err := stopAllJobs(cmd, appName, envName, cmpName); err != nil {
  				    return err
				}
			}
			
			if allBatches {
				if err :=stopAllBatches(cmd, appName, envName, cmpName); err != nil {
  				    return err
				}
			}
			return nil
```